phoenicis-cli/src/main/java/org/phoenicis/cli/scriptui/UiQuestionFactoryCLI.java

```diff
@@ -40,15 +55,33 @@ public void create(String questionText, Runnable yesCallback, Runnable noCallbac
 
             Scanner input = new Scanner(System.in);
             answer = input.nextLine();
+        }
+
+        return "yes".equals(answer);
+    }
 
-            switch (answer) {
-                case "yes":
-                    yesCallback.run();
-                    break;
-                case "no":
-                    noCallback.run();
-                    break;
+    @Override
+    public String create(String questionText, List<String> choices) {
```

Comment:
Don't we have another widget for this functionality already?

Reply:
We have something similar in the installation wizard, i.e.:

https://github.com/PhoenicisOrg/phoenicis/blob/master/phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/scriptui/StepRepresentationMenu.java

Which is used in

https://github.com/PhoenicisOrg/scripts/blob/c1bf3755e2c8cb6831695fb8ef98038e13fcd1a7/Applications/Games/League%20of%20Legends/Online/script.js#L34

The issue with this functionality is that it is contained inside the wizard, which can be quite easily overlooked when unexpected messages like a yes/no question for some unexpected situation is shown. A dialog that appears as a popup does not have this problem.

Reply:
Doesn't a dialog break the concept of the wizard? Also: If you have multiple open wizards, it will be very hard to figure out where the dialog belongs to.

Reply:
Both are true, but why do we have `UiQuestionFactory` then?

Reply:
I think we have two choices here, either:
- we remove the `UIQuestionFactory` and move all "question" dialogs inside the installation wizards. I think this would work, but could be a bit problematic in case a user overlooks a question. (is there perhaps a way to make alert the user of a change that requires user input in the wizard?)
- we support some categories of dialogs, e.g. question dialogs for now.

@plata @ImperatorS79 @Zemogiter @qparis what do you think?

Reply:
I'm against removing UIQuestionFactory because of the problems you mentioned. Also because we would have to change every script that uses it.

Reply:
If we keep it, we need a clear guideline when to use it and when not.

Reply:
I'm fine anyway but I think we should come to a decision relatively soon so that we can merge the blocked script PRs.

Reply:
@plata I don't think we really need a guideline for that because it's simply for asking questions to the user and letting said user choose from x options.

Reply:
@Zemogiter you could also use the menu step for that purpose.

However, don't feel blocked by this. We can still decide this in general later on and then change some scripts if required.